lib/internal/test_runner/utils.js

```diff
@@ -250,45 +263,130 @@ function countCompletedTest(test, harness = test.root.harness) {
 }
 
 
-function coverageThreshold(coverage, color) {
-  coverage = NumberPrototypeToFixed(coverage, 2);
-  if (color) {
-    if (coverage > 90) return `${green}${coverage}${color}`;
-    if (coverage < 50) return `${red}${coverage}${color}`;
-  }
-  return coverage;
+function addTableLine(prefix, width) {
+  return `${prefix}${'-'.repeat(width)}\n`;
+}
+
+function truncateStart(string, width) {
+  return string.length > width ? `\u2026${string.substring(string.length - width + 1, string.length)}` : string;
+}
+
+function truncateEnd(string, width) {
+  return string.length > width ? `${string.substring(0, width - 1)}\u2026` : string;
+}
+
+function formatLinesToRanges(values) {
+  return ArrayPrototypeMap(ArrayPrototypeReduce(values, (prev, current, index, array) => {
+    if ((index > 0) && ((current - array[index - 1]) === 1)) {
+      prev[prev.length - 1][1] = current;
+    } else {
+      prev.push([current]);
+    }
+    return prev;
+  }, []), (range) => range.join('-'));
+}
+
+function formatUncoveredLines(lines, table) {
+  if (table) return ArrayPrototypeJoin(formatLinesToRanges(lines), ' ');
+  return ArrayPrototypeJoin(lines, ', ');
 }
 
-function getCoverageReport(pad, summary, symbol, color) {
-  let report = `${color}${pad}${symbol}start of coverage report\n`;
+const kColumns = ['line %', 'branch %', 'funcs %'];
+const kColumnsKeys = ['coveredLinePercent', 'coveredBranchPercent', 'coveredFunctionPercent'];
+const kSeparator = ' | ';
+
+function getCoverageReport(pad, summary, symbol, color, table) {
+  const prefix = `${pad}${symbol}`;
+  let report = `${color}${prefix}start of coverage report\n`;
+
+  let filePadLength;
+  let columnPadLengths = [];
+  let uncoveredLinesPadLength;
+  let tableWidth;
+
+  if (table) {
+    // Get expected column sizes
+    filePadLength = table && ArrayPrototypeReduce(summary.files, (acc, file) =>
+      MathMax(acc, relative(summary.workingDirectory, file.path).length), 0);
+    filePadLength = MathMax(filePadLength, 'file'.length);
+    const fileWidth = filePadLength + 2;
+
+    columnPadLengths = ArrayPrototypeMap(kColumns, (column) => (table ? MathMax(column.length, 6) : 0));
+    const columnsWidth = ArrayPrototypeReduce(columnPadLengths, (acc, columnPadLength) => acc + columnPadLength + 3, 0);
+
+    uncoveredLinesPadLength = table && ArrayPrototypeReduce(summary.files, (acc, file) =>
+      MathMax(acc, formatUncoveredLines(file.uncoveredLineNumbers, table).length), 0);
+    uncoveredLinesPadLength = MathMax(uncoveredLinesPadLength, 'uncovered lines'.length);
+    const uncoveredLinesWidth = uncoveredLinesPadLength + 2;
+
+    tableWidth = fileWidth + columnsWidth + uncoveredLinesWidth;
+
+    // Fit with sensible defaults
+    const availableWidth = (process.stdout.columns || 9000) - prefix.length;
```

Comment:
why 9000? whenever I see a magic number I prefer some constant with a self explanatory name

Reply:
I thought I picked it up from somewhere else in the codebase but it was actually coming from a test. Defaulting to Infinity instead.